UPGRADING.md

```diff
@@ -0,0 +1,105 @@
+# Upgrading Guide
+
+See the [CHANGELOG.md](./CHANGELOG.md) for detailed information about what has changed between versions.
+
+This guide is useful to figure out what you need to do between breaking changes.
+
+As always, [submit issues](https://github.com/FortAwesome/vue-fontawesome/issues/new) that you run into with this guide or with these upgrades to us.
+
+## 0.0.x to 0.1.0
+
+### Renamed packages
+
+The following packages have been renamed as part of 5.1.0 of Font Awesome.
+
+_All packages are in the [@fortawesome NPM scope](https://www.npmjs.com/search?q=scope:fortawesome&page=1&ranking=optimal)_
+
+| Old package(1)           | New package            |
+|--------------------------|------------------------|
+| fontawesome              | fontawesome-svg-core   |
+| fontawesome-free-solid   | free-solid-svg-icons   |
+| fontawesome-free-regular | free-regular-svg-icons |
+| fontawesome-free-brands  | free-brands-svg-icons  |
+| fontawesome-pro-solid    | pro-solid-svg-icons    |
+| fontawesome-pro-regular  | pro-regular-svg-icons  |
+| fontawesome-pro-light    | pro-light-svg-icons    |
+
+(1) Old packages have now been deprecated. They are still available but will only receive high priority patch release fixes.
+
+### No more default imports
+
+Recently we spent a good deal of time supporting TypeScript to enable us to
+create the Angular Font Awesome component. During that adventure we
+[were](https://basarat.gitbooks.io/typescript/docs/tips/defaultIsBad.html)
+[convinced](https://blog.neufund.org/why-we-have-banned-default-exports-and-you-should-do-the-same-d51fdc2cf2ad)
+that we were going to remove default exports from all of our components,
+libraries, and packages. This is complete with the umbrella release of `5.1.0` of Font Awesome.
+
+What does that mean?
+
+~~Old way:~~
+
+```javascript
+import fontawesome from '@fontawesome/fontawesome'
+import solid from '@fortawesome/fontawesome-free-solid'
+import faTwitter from '@fortawesome/fontawesome-free-brands/faTwitter'
+import FontAwesomeIcon from '@fortawesome/vue-fontawesome'
+
+import fontaweome.library.add(solid, faTwitter)
+```
+
+New way:
+
+```javascript
+import { library } from '@fontawesome/fontawesome-svg-core'
+import { fas } from '@fortawesome/fontawesome-free-solid'
+import { faTwitter } from '@fortawesome/free-brands-svg-icons'
+import { FontAwesomeIcon } from '@fortawesome/vue-fontawesome'
+
+import library.add(fas, faTwitter)
+```
+
+### Improved support for tree shaking
+
+Tree shaking is now functional by default and no additional configuration is required to make it work.
+
+The `shakable.es.js` module has been removed and is no longer needed.
+
+If you've previously configured tree shaking by modifying your webpack or rollup you can safely remove these.
+
+**We recommend that you check your bundle size after upgrading an ensure that file sizes are as you would expect.**
```

Comment:
I think it would be helpful to suggest another check, since bundle size could seem unclear, if you're not sure what size to expect it to be when tree-shaken.

An alternative would be to look in the final bundle for an icon string like `faCoffee` that you didn't intend to use but which is otherwise defined in an icon content module with other icons that you _do intend_ to use.

For example, `@fortawesome/free-brands-svg-icons` includes both the icon named `twitter` (imported here into the JavaScript name `faTwitter`, and the icon named `google`. You should be able to search either the minified or un-minified bundle for those icon names and find an Icon Definition for `twitter` and _not_ for `google`.

Reply:
Yeah. You are correct. 😕 But this is so complex to explain. I'm hoping that we can roll the dice on this tree shaking stuff and hope that it Just Works (TM). Can I convince you to leave this vague for now and see what shakes out (pun intended)?